packages/utils/src/commonwell/doc-query-shared.ts

```diff
@@ -0,0 +1,95 @@
+import { MetriportMedicalApi } from "@metriport/api-sdk";
+import { sleep } from "@metriport/core/util/sleep";
+import axios from "axios";
+
+export type DetailedConfig = {
+  patientChunkDelayJitterMs: number;
+  queryPollDurationMs: number;
+  maxQueryDurationMs: number; // CW has a 70s timeout, so this is the maximum duration any doc query can take
+  maxDocQueryAttemts: number;
+  /**
+   * If the doc query returns less than this, we want to query again to try and get better
+   * coverage. Had situation where we requeried a patient w/ 1 doc ref and it jumped from
+   * 1 to 20.
+   */
+  minDocsToConsiderCompleted: number;
+};
+export type QueryParams = {
+  cxId: string;
+  patientId: string;
+  apiUrl: string;
+  apiKey: string;
+  config?: DetailedConfig;
+  log?: typeof console.log;
+};
+
+export async function queryDocsForPatient({
+  cxId,
+  patientId,
+  apiUrl,
+  apiKey,
+  config = {
+    patientChunkDelayJitterMs: 1000,
+    queryPollDurationMs: 10_000,
+    maxQueryDurationMs: 71_000,
+    maxDocQueryAttemts: 3,
+    minDocsToConsiderCompleted: 2,
+  },
+  log = console.log,
+}: QueryParams): Promise<{
+  docQueryAttempts: number;
+  docsFound: number;
+  queryComplete: boolean;
+}> {
+  const {
+    patientChunkDelayJitterMs,
+    queryPollDurationMs,
+    maxQueryDurationMs,
+    maxDocQueryAttemts,
+    minDocsToConsiderCompleted,
+  } = config;
+  let docQueryAttempts = 0;
+  let docsFound = 0;
+  let queryComplete = false;
+
+  const metriportAPI = new MetriportMedicalApi(apiKey, {
+    baseAddress: apiUrl,
+  });
```

Comment:
would leave a TODO for @jonahkaye to address under his PR related to the WH override https://github.com/metriport/metriport/pull/1106

Reply:
But we want to trigger WH in this case, right?

Reply:
Added a param